File: apps/mobile/lib/domain/usecases/class/get_class_by_id_usecase.dart
import '../../../core/usecases/usecase.dart';
import '../../../core/error/failures.dart';
import '../../../core/utils/either.dart';
import '../../entities/class/class_entity.dart';
import '../../repositories/school/class_repository.dart';

class GetClassByIdUseCase
    implements UseCase<Either<Failure, ClassEntity>, String> {
  final ClassRepository repository;

  GetClassByIdUseCase(this.repository);

  @override
  Future<Either<Failure, ClassEntity>> call(String params) async {
    return await repository.getClassById(params);
  }
}

File: services/user_service/src/domain/usecases/class/GetClassByIdUseCase.ts
import { IClassRepository } from "../../repositories/IClassRepository";
import { Class } from "../../entities/Class";

export class GetClassByIdUseCase {
  constructor(private classRepository: IClassRepository) {}

  async execute(id: string): Promise<Class | null> {
    return await this.classRepository.findById(id);
  }
}
